### PR TITLE
[soapysdr] New Port

### PR DESCRIPTION
**Pull request by howard0su**
- [x] Changes comply with the [maintainer guide](https://github.com/microsoft/vcpkg-docs/blob/main/vcpkg/contributing/maintainer-guide.md).
- [x] The name of the port matches an existing name for this component on https://repology.org/ if possible, and/or is strongly associated with that component on search engines.
- [x] Optional dependencies are resolved in exactly one way. For example, if the component is built with CMake, all `find_package` calls are REQUIRED, are satisfied by `vcpkg.json`'s declared dependencies, or disabled with [CMAKE_DISABLE_FIND_PACKAGE_Xxx](https://cmake.org/cmake/help/latest/variable/CMAKE_DISABLE_FIND_PACKAGE_PackageName.html).
- [x] The versioning scheme in `vcpkg.json` matches what upstream says.
- [x] The license declaration in `vcpkg.json` matches what upstream says.
- [x] The installed as the "copyright" file matches what upstream says.
- [x] The source code of the component installed comes from an authoritative source.
- [x] The generated "usage text" is accurate. See [adding-usage](https://github.com/microsoft/vcpkg-docs/blob/main/vcpkg/examples/adding-usage.md) for context.
- [x] The version database is fixed by rerunning `./vcpkg x-add-version --all` and committing the result.
- [x] Only one version is in the new port's versions file.
- [x] Only one version is added to each modified port's versions file.

### CODE REVIEW

**Comment by howard0su** [comment]
> @microsoft-github-policy-service agree [company="{your company}"]

@microsoft-github-policy-service agree [company="Microsoft"]

**Comment by howard0su** [comment]
Not ready

File: ports/soapysdr/portfile.cmake [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+    FILE_LIST "${SOURCE_PATH}/LICENSE_1_0.txt"
+)
+
+
+file(REMOVE_RECURSE "${CURRENT_PACKAGES_DIR}/debug/include")
+
+if (VCPKG_TARGET_IS_WINDOWS)
+file(INSTALL ${CURRENT_PACKAGES_DIR}/cmake/ DESTINATION ${CURRENT_PACKAGES_DIR}/share/soapysdr/)
+file(REMOVE_RECURSE "${CURRENT_PACKAGES_DIR}/cmake")
+endif()
```
Comment (LilyWangLL) [inline ports/soapysdr/portfile.cmake:30]:
```suggestion
vcpkg_cmake_config_fixup(CONFIG_PATH cmake)
```

File: ports/soapysdr/portfile.cmake [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+
+file(REMOVE_RECURSE "${CURRENT_PACKAGES_DIR}/debug/cmake")
```
Comment (LilyWangLL) [inline ports/soapysdr/portfile.cmake:32]:
```suggestion
```

File: ports/soapysdr/vcpkg.json [hunk 3]
```diff
@@ -0,0 +1,18 @@
+{
+  "name": "soapysdr",
+  "version": "0.8.1",
+  "port-version": 1,
+  "description": "Vendor and platform neutral SDR interface library.",
+  "homepage": "https://github.com/pothosware/SoapySDR/wiki",
```
Comment (LilyWangLL) [inline ports/soapysdr/vcpkg.json:6]:
```suggestion
  "homepage": "https://github.com/pothosware/SoapySDR/wiki",
  "license": "BSL-1.0",
```

**Review by LilyWangLL — CHANGES_REQUESTED** [review]

**Comment by LilyWangLL** [comment]
Usage test passed on `x64-windows`.

File: ports/soapysdr/portfile.cmake [hunk 4]
```diff
@@ -0,0 +1,35 @@
+vcpkg_from_github(
+    OUT_SOURCE_PATH SOURCE_PATH
+    REPO pothosware/SoapySDR
+    REF soapy-sdr-0.8.1
```
Comment (dg0yt) [inline ports/soapysdr/portfile.cmake:4]:
```suggestion
    REF soapy-sdr-${VERSION}
```

File: ports/soapysdr/portfile.cmake [hunk 5]
```diff
@@ -0,0 +1,35 @@
...
+    SHA512 50c8f1652bf9ca09215f9c4115175427ca7b5338add7591e31ca0e627093c94b73e7cf7f84fa71ff419cc010d3c1263931506c728bbaa00413a7915d56a87787
+)
+
+vcpkg_cmake_configure(
+    SOURCE_PATH ${SOURCE_PATH}
```
Comment (dg0yt) [inline ports/soapysdr/portfile.cmake:9]:
```suggestion
    SOURCE_PATH "${SOURCE_PATH}"
```

File: ports/soapysdr/portfile.cmake [hunk 6]
```diff
@@ -0,0 +1,35 @@
...
+    OPTIONS
+        -DENABLE_DOCS=OFF
+        -DENABLE_APPS=OFF
+	-DENABLE_PYTHON=OFF
+	-DENABLE_PYTHON3=OFF
+	-DENABLE_TESTS=OFF
```
Comment (dg0yt) [inline ports/soapysdr/portfile.cmake:15]:
```suggestion
    OPTIONS
        -DENABLE_DOCS=OFF
        -DENABLE_APPS=OFF
        -DENABLE_PYTHON=OFF
        -DENABLE_PYTHON3=OFF
        -DENABLE_TESTS=OFF
```

File: ports/soapysdr/portfile.cmake [hunk 7]
```diff
@@ -0,0 +1,35 @@
...
+)
+
+vcpkg_cmake_install()
+
+vcpkg_install_copyright(
+    FILE_LIST "${SOURCE_PATH}/LICENSE_1_0.txt"
+)
```
Comment (dg0yt) [inline ports/soapysdr/portfile.cmake:22]:
```suggestion
vcpkg_install_copyright(FILE_LIST "${SOURCE_PATH}/LICENSE_1_0.txt")
```

**Review by LilyWangLL — DISMISSED** [review]

File: ports/soapysdr/portfile.cmake [hunk 8]
```diff
@@ -0,0 +1,27 @@
...
+vcpkg_cmake_install()
+
+vcpkg_install_copyright(FILE_LIST "${SOURCE_PATH}/LICENSE_1_0.txt")
+
+vcpkg_cmake_config_fixup(CONFIG_PATH cmake)
+
+file(REMOVE_RECURSE "${CURRENT_PACKAGES_DIR}/debug/include")
+file(REMOVE_RECURSE "${CURRENT_PACKAGES_DIR}/debug/share")
+
+vcpkg_fixup_pkgconfig()
```
Comment (dg0yt) [inline ports/soapysdr/portfile.cmake:27]: Nice to have, more common pattern:
```suggestion
vcpkg_cmake_install()
vcpkg_fixup_pkgconfig()
vcpkg_cmake_config_fixup(CONFIG_PATH cmake)

file(REMOVE_RECURSE "${CURRENT_PACKAGES_DIR}/debug/include")
file(REMOVE_RECURSE "${CURRENT_PACKAGES_DIR}/debug/share")

vcpkg_install_copyright(FILE_LIST "${SOURCE_PATH}/LICENSE_1_0.txt")
```

File: ports/soapysdr/vcpkg.json [hunk 9]
```diff
@@ -0,0 +1,19 @@
+{
+  "name": "soapysdr",
+  "version": "0.8.1",
+  "port-version": 1,
```
Comment (JavierMatosD) [inline ports/soapysdr/vcpkg.json:4]:
```suggestion
```

This is a new port. The `port-version`, should only be bumped between `version` updates.

**Review by JavierMatosD — CHANGES_REQUESTED** [review]

**Comment by JavierMatosD** [comment]
Mark ready for review when you've applied the changes

File: ports/soapysdr/usage [hunk 10]
```diff
@@ -0,0 +1,9 @@
+soapysdr provides CMake targets:
+
+  find_package(SoapySDR CONFIG REQUIRED)
+  target_link_libraries(main PRIVATE SoapySDR)
+
+soapysdr provides pkg-config modules:
+
+  # Vendor and platform neutral SDR interface library.
+  SoapySDR
```
Comment (dg0yt) [inline ports/soapysdr/usage:9]:
```suggestion
  SoapySDR
```
The last line must end with a line break.

**Review by JavierMatosD — APPROVED** [review]

**Pull request merged**